### ISSUE TITLE
Field interface

### ISSUE THREAD

**Issue by pinonpierre** [vaadin/framework#9519]
When I manipulate list of fields I often need to cast fields because all fields have not common interface/class except HasValue but HasValue is not a component.

The idea is to have Field that inherit HasValue + Component
A lot of fields inherit of AbstractField but not all...

**Comment by tsuoanttila** [comment]
Some features of internal implementation of AbstractField don't apply to everything. If you have for example a method that wants to handle something that is a HasValue and a Component at same time, you can use JavaGenerics to make that happen. Adding `<F extends HasValue<T> & Component>` makes a combination of `HasValue` and `Component` API called `F`. You can either provide the value type `T` as something common in a bigger scope, or add the `T` type to the generic as well.

**Comment by pinonpierre** [comment]
Yes, but when you need to store fields of different types in a List or Map, you can't choose a type because the types in List/Map are differents.

I need to define something like that:
`List<Field> myList = new ArraysList<>()`

But Field interface doesn't exists anymore in Vaadin 8.
and I can't define:
1/
`List<? extends HasValue<?> & Component> myList = new ArraysList<>()`
because it's not possible in Java to define "&" outside of generic parameter

and I can't define too:
2/
```
public class Test<FIELD extends HasValue<TYPE> & Component, TYPE> {
    private List<FIELD> myList = new ArrayList<>();
}
```
because it's not a list of unknown field type! It's a list of different field types!

Except this issue, please note that your design add a lot of complexity for nothing...


The best design to solve this problem is to create a new interface that inherits HasValue + Component
